temporal-sdk/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -1191,7 +1181,13 @@ private void processFailWorkflowExecution(
     }
 
     if (!Strings.isNullOrEmpty(data.cronSchedule)) {
-      startNewCronRun(ctx, workflowTaskCompletedId, identity, data, data.lastCompletionResult);
+      startNewCronRun(
+          ctx,
+          workflowTaskCompletedId,
+          identity,
+          data,
+          data.lastCompletionResult,
+          Optional.of(d.getFailure()));
```

Comment:
gRPC doesn't support nulls.
So d.getFailure never returns null and Optional.of of "d" that has d.hasFailure==false will be an optional of an empty Failure structure.